src/client/opamAdminCommand.ml

```diff
@@ -16,13 +16,33 @@ open Cmdliner
 
 type command = unit Cmdliner.Term.t * Cmdliner.Cmd.info
 
-let checked_repo_root () =
+let repo_version_lt repo_root v =
+  let v' =
+    let open OpamStd.Option.Op in
+    (OpamFile.Repo.read_opt
+       (OpamRepositoryPath.repo repo_root)
+     >>= OpamFile.Repo.opam_version)
+    +! (OpamVersion.of_string "1.2")
+  in
+  if OpamVersion.compare v v' > 0 then Some v' else None
+
+let checked_repo_root ?(check=true)() =
   let repo_root = OpamFilename.cwd () in
   if not (OpamFilename.exists_dir (OpamRepositoryPath.packages_dir repo_root))
   then
     OpamConsole.error_and_exit `Bad_arguments
       "No repository found in current directory.\n\
        Please make sure there is a \"packages%s\" directory" OpamArg.dir_sep;
+  (if check then
+     let default_current = "2.0" in
+     match repo_version_lt repo_root
+             (OpamVersion.of_string default_current) with
+     | Some v ->
+       OpamConsole.warning
```

Comment:
I think opam admin should simply refuse to handle 1.2 repositories
```suggestion
       OpamConsole.error_and_exit
```

Reply:
It can be done that way too. It need a proper cleaning of the code if there is some that handles 1.2 backward compatibility (expect upgrade ofc).